backend/llm/openai.py

```diff
@@ -2,136 +2,119 @@
 import json
 from typing import AsyncIterable, Awaitable
 
-from langchain.callbacks import AsyncIteratorCallbackHandler
-
-# Importing various modules and classes from a custom library 'langchain' likely used for natural language processing
 from langchain.chains import ConversationalRetrievalChain, LLMChain
 from langchain.chains.question_answering import load_qa_chain
 from langchain.chat_models import ChatOpenAI
 from langchain.embeddings.openai import OpenAIEmbeddings
 from langchain.llms.base import LLM
-from llm.prompt.CONDENSE_PROMPT import CONDENSE_QUESTION_PROMPT
 from logger import get_logger
-from models.settings import BrainSettings  # Importing settings related to the 'brain'
-from pydantic import BaseModel  # For data validation and settings management
+from models.chat import ChatHistory
+from models.settings import BrainSettings
 from repository.chat.get_chat_history import get_chat_history
 from repository.chat.update_chat_history import update_chat_history
 from repository.chat.update_message_by_id import update_message_by_id
-from supabase import Client  # For interacting with Supabase database
-from supabase import create_client
+from supabase import Client, create_client
 from vectorstore.supabase import (
     CustomSupabaseVectorStore,
 )  # Custom class for handling vector storage with Supabase
 
+from .base import BaseBrainPicking
+from .prompts.CONDENSE_PROMPT import CONDENSE_QUESTION_PROMPT
+
 logger = get_logger(__name__)
 
 
-class BrainPicking(BaseModel):
+class OpenAIBrainPicking(BaseBrainPicking):
     """
     Main class for the Brain Picking functionality.
     It allows to initialize a Chat model, generate questions and retrieve answers using ConversationalRetrievalChain.
     """
 
     # Instantiate settings
-    settings = BrainSettings()
+    brain_settings = BrainSettings()
 
     # Default class attributes
-    llm_name: str = "gpt-3.5-turbo"
+    model_name: str = "gpt-3.5-turbo"
     temperature: float = 0.0
+    brain_id: str
     chat_id: str
     max_tokens: int = 256
-
-    # Storage
-    supabase_client: Client = None
-    vector_store: CustomSupabaseVectorStore = None
-
-    # Language models
-    embeddings: OpenAIEmbeddings = None
-    question_llm: LLM = None
-    doc_llm: LLM = None
-    question_generator: LLMChain = None
-    doc_chain: LLMChain = None
-    qa: ConversationalRetrievalChain = None
-
-    # Streaming
-    callback: AsyncIteratorCallbackHandler = None
     streaming: bool = False
```

Comment:
already have this in BaseBrainPicking,no need to redefine it here, no ?

Reply:
Fixed